### DISCUSSION TITLE
How to automatically add newly created content definitions to `Audit Trail -> Allowed content types`?

### DISCUSSION THREAD

**Discussion by hieucd04** [OrchardCMS/OrchardCore#16438] [Q&A] [answered]
### What I did:
  1. Create a new `Content Definition` called `Test`
  2. Go to `Configurations -> Audit Trail`
 
### What happens:
  - The newly created `Content Definition` (`Test`) is not ticked (see the image below)

  ![bug](https://github.com/user-attachments/assets/264bd36c-7ab4-44e6-b505-b1d8b99ee2ff)

### What I'm trying to achieve:
  - The newly created `Content Definition` (`Test`) is automatically ticked without me having to do it manually.

Is there anyway to do it? (Preferably via a Setup Recipe with a note that I don't know the name of the content definitions, I want to add all)

**Comment by Piedone** [comment, marked as answer]
You'll need to do some coding for this, if the goal is to have all content types selected there, past and future, and you don't know their names in advance.

Some pointers (look at the source code of Audit Trail by cloning the OC repo): these are stored in `ContentAuditTrailSettings`. You update them from the UI using `ContentAuditTrailSettingsDisplayDriver`. You could implement an `IContentDefinitionEventHandler` and in `ContentTypeCreated` update `ContentAuditTrailSettings`.
